examples/legacy/seq2seq/seq2seq_trainer.py

```diff
@@ -135,7 +135,7 @@ def _get_lr_scheduler(self, num_training_steps):
     def _get_train_sampler(self) -> Optional[torch.utils.data.Sampler]:
         if isinstance(self.train_dataset, torch.utils.data.IterableDataset):
             return None
-        elif is_torch_tpu_available():
+        elif is_torch_xla_available():
```

Comment:
This isn't equivalent, previously, we were checking for a device, but by default that isn't happening anymore

Reply:
This is to align with [PR](https://github.com/huggingface/accelerate/pull/2467) of the accelerate library. If users do not wish to use torch_xla in an environment where torch_xla is installed, they can configure it using USE_TORCH_XLA, which is also the purpose of this PR.